service-monitor.yaml

```diff
@@ -0,0 +1,53 @@
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: opni-metrics-service
+  labels:
+    app.kubernetes.io/name: opni-svc-metrics
+    app.kubernetes.io/part-of: opni
+    opni.io/service: metrics
+spec:
+  type: ClusterIP
+  ports:
+  - name: metrics
+    port: 8000
+    targetPort: 8000
+    protocol: TCP
+  selector:
+    app.kubernetes.io/name: opni-svc-metrics
+    app.kubernetes.io/part-of: opni
+    opni.io/service: metrics
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: opni-metrics-service
+spec:
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: opni-svc-metrics
+      app.kubernetes.io/part-of: opni
+      opni.io/service: metrics
+  endpoints:
+  - port: metrics
+---
```

Comment:
just wondering , what does this yaml file do?

Reply:
This file sets up the prometheus resources to make sure the metrics service predictions get scraped.